src/Conf.ml

```diff
@@ -377,6 +385,45 @@ end = struct
     store := Pack opt :: !store ;
     opt
 
+  let opt ~type_ ~type_name ~of_string ~to_string ~docv ~names ~doc ~section
```

Comment:
Would be great if this could take any `option_decl`: `val opt : 'a option_decl -> 'a option option_decl`.
Doc, parsing function and `Arg.t` should compose, the problem is the way the `term` is wrapped in a function.

Reply:
I agree it would be nice, the whole Conf file could use a redesign but preferably in another PR.